azurecompute/src/test/java/org/jclouds/azurecompute/compute/AzureTemplateBuilderLiveTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.compute;
+
+import java.util.Set;
+import org.jclouds.azurecompute.AzureComputeProviderMetadata;
+import org.jclouds.compute.internal.BaseTemplateBuilderLiveTest;
+import org.jclouds.providers.ProviderMetadata;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "AzureTemplateBuilderLiveTest")
+public class AzureTemplateBuilderLiveTest extends BaseTemplateBuilderLiveTest {
+
+   private AzureComputeProviderMetadata providerMeta;
+
+   public AzureTemplateBuilderLiveTest() {
+      super();
+
+      provider = "azurecompute";
+   }
+
+   @Override
+   protected ProviderMetadata createProviderMetadata() {
+      synchronized (this) {
+         if (providerMeta == null) {
+            providerMeta = new AzureComputeProviderMetadata();
+         }
+      }
+      return providerMeta;
+   }
```

Comment:
I could not find any way to fetch the provider metadata once created by base class method (and I needed it for `getIso3166Codes()`), so I thought the simpler way was to override and provide a local reference.

BTW This is what I've also seen from [AWSEC2](https://github.com/jclouds/jclouds/blob/acd06b30245c4df27e4e36176916c95df60e42ca/providers/aws-ec2/src/test/java/org/jclouds/aws/ec2/compute/AWSEC2TemplateBuilderLiveTest.java#L65)